File: JAVA/3158. Find the XOR of Numbers Which Appear Twice.java
class Solution {
    public int duplicateNumbersXOR(int[] n) {
        int c = 0;
        Map<Integer, Integer> m = new HashMap<>();
        for (int i : n) {
            m.put(i, m.getOrDefault(i, 0) + 1);
            if (m.get(i) == 2) c ^= i;
        }
        return c;
    }
}

File: C++/3158. Find the XOR of Numbers Which Appear Twice.cpp
class Solution {
public:
    int duplicateNumbersXOR(vector<int>& n) {
        int c=0;
        map<int,int>m;
        for(int i:n){
            m[i]++;
            if(m[i]==2)
            c^=i;
        }
        return c;
    }
};